### PR TITLE
[#130, #134] - merge FilterInstruction to Instruction and rename HyperledgerListenerHelper to HyperledgerInstructionHelper

### LINKED ISSUE #130: Merge FilterInstruction and Instruction into a single Instruction abstract class

### LINKED ISSUE #134: Rename HyperledgerListenerHelper  to HyperledgerInstructionHelper

### PR DESCRIPTION
**Pull request by mykola-digtiar**
Closes #130. Closes #134.

Instead of merging everything into one Instruction interface, it is better to declare an Instruction as an abstract class since the executeNestedInstrction method does not have to be implemented in every instruction class but in the case of an interface, this would be impossible.

Implementing the Interface as an abstract class gives a possibility to bake in the reusable functionality (`executeNestedInstructions` method in particular) while giving freedom to a developer to override everything if needed. By default, the `execute` method does nothing (if not overridden).

The Jenkins tests [succeeded](https://elf-ci.rohrschacht.de/job/BLF/247/).

### CODE REVIEW

**Comment by Neckster** [comment]
Fixed last commit message.

**Review by Knostpe — DISMISSED** [review]

File: src/main/java/blf/blockchains/ethereum/instructions/EthereumTransactionFilterInstruction.java [hunk 1]
```diff
@@ -39,28 +38,33 @@ public EthereumTransactionFilterInstruction(
...
+                        ethereumReader.setCurrentTransaction(tx);
+                        this.executeNestedInstructions(state);
+                    } catch (Exception cause) {
+                        final String message = String.format(
+                            "Error mapping transaction '%s' in block '%s'.",
+                            tx.getTransactionIndex(),
+                            tx.getBlockNumber()
+                        );
+                        final boolean abort = state.getExceptionHandler().handleExceptionAndDecideOnAbort(message, cause);
+                        if (abort) {
```
Comment (rohrschacht) [inline src/main/java/blf/blockchains/ethereum/instructions/EthereumTransactionFilterInstruction.java]: I don't believe this is necessary anymore. Isn't the plan to make handleExceptionAndDecideOnAbort directly abort the program and not return a boolean anymore? Also, if it returns true, you just call it again.

**Review by rohrschacht — COMMENTED** [review]
This is great work as always! And you even also started adding the global exceptionHandler everywhere. I had just one small comment about using a returned value from the exceptionHandler, where I am not sure that it is still relevant.

File: src/main/java/blf/blockchains/ethereum/instructions/EthereumTransactionFilterInstruction.java (hunk 1 above)
Comment (mykola-digtiar) [reply src/main/java/blf/blockchains/ethereum/instructions/EthereumTransactionFilterInstruction.java]: You are right!

**Review by rohrschacht — APPROVED** [review]
The code looks good to me and Jenkins approved as well :+1: Nice work!

**Review by Knostpe — APPROVED** [review]

**Pull request merged**